apps/call_response.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package apps
+
+import (
+	"fmt"
+)
+
+type CallResponseType string
+
+const (
+	// CallResponseTypeOK indicates that the call succeeded, returns optional
+	// Markdown (message) and Data.
+	CallResponseTypeOK CallResponseType = "ok"
+
+	// CallResponseTypeOK indicates an error, returns ErrorText, and optional
+	// field-level errors in Data.
+	CallResponseTypeError CallResponseType = "error"
+
+	// CallResponseTypeForm returns Form, the definition of the form to display.
+	// If returned responding to a submit, causes the form to be displayed as a
+	// modal.
+	CallResponseTypeForm CallResponseType = "form"
+
+	// CallResponseTypeCall indicates that another Call that should be executed
+	// (all the way from the user-agent). Call is returned. NOT YET IMPLEMENTED.
+	CallResponseTypeCall CallResponseType = "call"
+
+	// CallResponseTypeNavigate indicates that the user should be forcefully
+	// navigated to a URL, which may be a channel in Mattermost. NavigateToURL
+	// and UseExternalBrowser are expected to be returned.
+	CallResponseTypeNavigate CallResponseType = "navigate"
+)
+
+// CallResponse is general envelope for all Call responses.
+//
+// Submit requests expect ok, error, form, call, or navigate response types.
+// Returning a "form" type in response to a submission from the user-agent
+// triggers displaying a Modal. Returning a "call" type in response to a
+// submission causes the call to be executed from the user-agent (NOT
+// IMPLEMENTED YET)
+//
+// Form requests expect form or error.
+//
+// Lookup requests expect ok or error.
+//
+// In case of an error, the returned response type is "error", ErrorText
+// contains the overall error text. Data contains optional, field-level errors.
+type CallResponse struct {
+	Type CallResponseType `json:"type"`
+
+	// Text is used for OK and Error response, and will show the text in the
+	// proper output.
```

Comment:
I'm actually wondering if we should keep ` Error` as a field as all other call response types use there own field, why not OK and Error?

Reply:
The main idea was to simplify the structures. @lev thoughts?

Reply:
Well, `Error` was a `string` before and as such does not need a special field. If we decide to add data to it, we can do it as `Data`, as per convention.